# SPDX-License-Identifier: Apache-2.0

--- fcct-config.yaml	2022-11-23 14:59:24.000000000 +0000
+++ fcct-config.yaml.modified	2023-06-15 17:51:13.784041161 +0000
@@ -56,7 +56,7 @@
           #     mls - Multi Level Security protection.
           SELINUXTYPE=targeted
       overwrite: true
-    - path: /etc/containers/libpod.conf
+    - path: /etc/containers/containers.conf
       # 420 (decimal) == 644 (octal)
       mode: 420
       user:
@@ -65,10 +65,11 @@
         name: root
       contents:
         inline: |
+        [containers]
           # Maximum size of log files (in bytes)
           # -1 is unlimited
           # 50m
-          max_log_size = 52428800
+          log_size_max = 52428800
     - path: /etc/containers/__REGISTRIES_CONF__
       # 420 (decimal) == 644 (octal)
       mode: 420
