--- ../original/config.py	2020-03-27 15:57:26.130007550 -0500
+++ ./config.py	2020-03-27 15:57:26.130007550 -0500
@@ -102,6 +102,8 @@
                help=_('Iptables mangle mark used to mark ingress from '
                       'external network. This mark will be masked with '
                       '0xffff so that only the lower 16 bits will be used.')),
+    cfg.StrOpt('dmz_cidr', default='',
+               help=_('WMF defined src nat exclusions "src_range:dst_range,<repeat>')),
 ]
 
 OPTS += config.EXT_NET_BRIDGE_OPTS
