# SPDX-License-Identifier: Apache-2.0

--- models.py	2022-09-14 14:10:06.000000000 +0000
+++ models.py.patched	2023-01-11 14:45:30.038446481 +0000
@@ -64,6 +64,7 @@
 from trove.instance.models import Instance
 from trove.instance.models import InstanceServiceStatus
 from trove.instance.models import InstanceStatus
+from trove.instance.models import load_simple_instance_addresses
 from trove.instance.tasks import InstanceTasks
 from trove.module import models as module_models
 from trove.module import views as module_views
@@ -1057,9 +1058,7 @@
 
             utils.poll_until(get_server, ip_is_available,
                              sleep_time=1, time_out=CONF.dns_time_out)
-            server = self.nova_client.servers.get(
-                self.db_info.compute_instance_id)
-            self.db_info.addresses = server.addresses
+            load_simple_instance_addresses(self.context, self.db_info)
             LOG.debug("Creating dns entry...")
             ip = self.dns_ip_address
             if not ip:
