# SPDX-License-Identifier: Apache-2.0
--- cinder/scheduler/manager.py	2023-01-24 23:07:10.000000000 +0000
+++ cinder/scheduler/manager.py.patched	2023-04-16 23:49:10.705379295 +0000
@@ -638,9 +638,10 @@
         volume_id = backup.volume_id
         volume = self.db.volume_get(context, volume_id)
         try:
-            host = self.driver.get_backup_host(volume)
-            backup.host = host
-            backup.save()
+            if not backup.host:
+                host = self.driver.get_backup_host(volume)
+                backup.host = host
+                backup.save()
             self.backup_api.create_backup(context, backup)
         except exception.ServiceNotFound:
             self.db.volume_update(context, volume_id,
