# SPDX-License-Identifier: Apache-2.0

--- start-container-agent.sh	2022-11-23 14:59:24.000000000 +0000
+++ start-container-agent.sh.modified	2023-06-15 17:48:31.597520059 +0000
@@ -53,11 +53,12 @@
 _prefix="${CONTAINER_INFRA_PREFIX:-docker.io/openstackmagnum/}"
 
 if [ "$(echo $USE_PODMAN | tr '[:upper:]' '[:lower:]')" == "true" ]; then
-    cat > /etc/containers/libpod.conf <<EOF
+    cat > /etc/containers/containers.conf <<EOF
+[containers]
 # Maximum size of log files (in bytes)
 # -1 is unlimited
 # 50m
-max_log_size = 52428800
+log_size_max = 52428800
 EOF
     cat > /etc/systemd/system/heat-container-agent.service <<EOF
 [Unit]
