# SPDX-License-Identifier: Apache-2.0

diff --git a/keystone/trust/schema.py b/keystone/trust/schema.py
index 20c85b390..90db3fafd 100644
--- a/keystone/trust/schema.py
+++ b/keystone/trust/schema.py
@@ -61,7 +61,6 @@ _trust_properties = {
     },
     "project_id": {
         "type": ["string", "null"],
-        "format": "uuid",
         "description": (
             "Identifies the project upon which the trustor is "
             "delegating authorization."
-- 
2.39.5

