# SPDX-License-Identifier: Apache-2.0

--- keystone/common/utils.py	2023-07-25 12:36:29.702067459 -0500
+++ keystone/common/utils.py.patched	2023-07-25 12:39:21.495784897 -0500
@@ -356,7 +356,16 @@
 
 def is_not_url_safe(name):
     """Check if a string contains any url reserved characters."""
-    return len(list_url_unsafe_chars(name)) > 0
+
+    # HACK ALERT
+    #
+    # This is only used to vet domain and project names.
+    #
+    # Reject anything that's not alpha-numeric. This is to ensure
+    #  compatibility with radosgw. T341509
+    #
+    #
+    return (not name.isalnum())
 
 
 def list_url_unsafe_chars(name):
