# SPDX-License-Identifier: Apache-2.0

diff --git a/keystone/identity/schema.py b/keystone/identity/schema.py
index f8688baae..81a28fb82 100644
--- a/keystone/identity/schema.py
+++ b/keystone/identity/schema.py
@@ -87,7 +87,7 @@ user_index_request_query: dict[str, Any] = {
 }
 
 _user_properties: dict[str, Any] = {
-    'id': {"type": "string", "description": "The user ID.", "readOnly": True},
+    'id': parameter_types.user_id,
     'default_project_id': validation.nullable(parameter_types.id_string),
     'description': validation.nullable(parameter_types.description),
     'domain_id': parameter_types.id_string,
