# SPDX-License-Identifier: Apache-2.0

--- enginefacade.py	2024-08-25 15:32:31.368020590 -0500
+++ enginefacade.py.modified	2024-08-25 15:39:49.088183191 -0500
@@ -176,6 +176,7 @@
         }
         self._maker_cfg = {
             'expire_on_commit': _Default(False),
+            '__autocommit': False,
         }
         self._transaction_ctx_cfg = {
             'rollback_reader_sessions': False,
@@ -446,6 +447,7 @@
 
     def _maker_args_for_conf(self, conf):
         maker_args = self._args_for_conf(self._maker_cfg, conf)
+        maker_args['autocommit'] = maker_args.pop('__autocommit')
         return maker_args
 
     def dispose_pool(self):
@@ -1214,6 +1216,9 @@
     :param sqlite_fk: enable foreign keys in SQLite
     :type sqlite_fk: bool
 
+    :param autocommit: use autocommit mode for created Session instances
+    :type autocommit: bool
+
     :param expire_on_commit: expire session objects on commit
     :type expire_on_commit: bool
 
@@ -1251,14 +1256,22 @@
 
     """
     def __init__(self, sql_connection, slave_connection=None,
-                 sqlite_fk=False, expire_on_commit=False, _conf=None,
-                 _factory=None, **kwargs):
+                 sqlite_fk=False, autocommit=False,
+                 expire_on_commit=False, _conf=None, _factory=None, **kwargs):
         warnings.warn(
             "EngineFacade is deprecated; please use "
             "oslo_db.sqlalchemy.enginefacade",
             warning.OsloDBDeprecationWarning,
             stacklevel=2)
 
+        if autocommit is True:
+            warnings.warn(
+                'autocommit support will be removed in SQLAlchemy 2.0 and '
+                'should not be relied on; please rework your code to remove '
+                'reliance on this feature',
+                warning.OsloDBDeprecationWarning,
+                stacklevel=2)
+
         if _factory:
             self._factory = _factory
         else:
@@ -1266,6 +1279,7 @@
 
             self._factory.configure(
                 sqlite_fk=sqlite_fk,
+                __autocommit=autocommit,
                 expire_on_commit=expire_on_commit,
                 **kwargs
             )
@@ -1331,7 +1345,7 @@
 
     @classmethod
     def from_config(cls, conf,
-                    sqlite_fk=False, expire_on_commit=False):
+                    sqlite_fk=False, autocommit=False, expire_on_commit=False):
         """Initialize EngineFacade using oslo.config config instance options.
 
         :param conf: oslo.config config instance
@@ -1340,6 +1354,9 @@
         :param sqlite_fk: enable foreign keys in SQLite
         :type sqlite_fk: bool
 
+        :param autocommit: use autocommit mode for created Session instances
+        :type autocommit: bool
+
         :param expire_on_commit: expire session objects on commit
         :type expire_on_commit: bool
 
@@ -1348,4 +1365,5 @@
         return cls(
             None,
             sqlite_fk=sqlite_fk,
+            autocommit=autocommit,
             expire_on_commit=expire_on_commit, _conf=conf)
