# SPDX-License-Identifier: Apache-2.0

--- cinder/api/schemas/volume_type_access.py	2022-06-01 15:54:33.000000000 +0000
+++ cinder/api/schemas/volume_type_access.py.modified	2022-11-15 21:28:22.104310972 +0000
@@ -27,7 +27,7 @@
         'addProjectAccess': {
             'type': 'object',
             'properties': {
-                'project': parameter_types.uuid,
+                'project': parameter_types.nullable_string,
             },
             'required': ['project'],
             'additionalProperties': False,
@@ -45,7 +45,7 @@
         'removeProjectAccess': {
             'type': 'object',
             'properties': {
-                'project': parameter_types.uuid,
+                'project': parameter_types.nullable_string,
             },
             'required': ['project'],
             'additionalProperties': False,
