# SPDX-License-Identifier: Apache-2.0
--- cinder/api/schemas/volume_type_access.py	2022-11-28 11:12:42.657807587 -0600
+++ cinder/api/schemas/volume_type_access.py.modified	2022-11-28 11:12:37.089809658 -0600
@@ -18,8 +18,6 @@
 
 """
 
-from cinder.api.validation import parameter_types
-
 add_project_access = {
     'type': 'object',
     'properties': {
@@ -27,7 +25,8 @@
         'addProjectAccess': {
             'type': 'object',
             'properties': {
-                'project': parameter_types.uuid,
+                'project': {'type': 'string',
+                            'minLength': 1, 'maxLength': 255},
             },
             'required': ['project'],
             'additionalProperties': False,
@@ -45,7 +44,8 @@
         'removeProjectAccess': {
             'type': 'object',
             'properties': {
-                'project': parameter_types.uuid,
+                'project': {'type': 'string',
+                            'minLength': 1, 'maxLength': 255},
             },
             'required': ['project'],
             'additionalProperties': False,
